app/src/components/form/LModal.vue

```diff
@@ -0,0 +1,28 @@
+<script setup lang="ts">
+type Props = {
+    isVisible: boolean;
+    heading: string;
+};
+defineProps<Props>();
+
+const emit = defineEmits(["close"]);
+</script>
+
+<template>
+    <Teleport to="body">
+        <div v-if="isVisible" @click.self="emit('close')">
+            <div class="fixed inset-0 z-50 bg-slate-800 bg-opacity-50 backdrop-blur-sm"></div>
```

Comment:
In light mode we should perhaps choose a zinc style colour?
```suggestion
            <div class="fixed inset-0 z-50 bg-zink-800 dark:bg-slate-800 bg-opacity-50 backdrop-blur-sm"></div>
```